spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/config/DataFlowControllerAutoConfiguration.java

```diff
@@ -0,0 +1,156 @@
+ */
+
+package org.springframework.cloud.dataflow.server.config;
+
+import org.springframework.boot.autoconfigure.condition.ConditionalOnBean;
+import org.springframework.boot.autoconfigure.security.SecurityProperties;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.cloud.dataflow.app.resolver.ModuleResolver;
+import org.springframework.cloud.dataflow.artifact.registry.ArtifactRegistry;
+import org.springframework.cloud.dataflow.completion.StreamCompletionProvider;
+import org.springframework.cloud.dataflow.server.controller.CompletionController;
+import org.springframework.cloud.dataflow.server.controller.CounterController;
+import org.springframework.cloud.dataflow.server.controller.FieldValueCounterController;
+import org.springframework.cloud.dataflow.server.controller.LibraryController;
+import org.springframework.cloud.dataflow.server.controller.ModuleController;
+import org.springframework.cloud.dataflow.server.controller.RestControllerAdvice;
+import org.springframework.cloud.dataflow.server.controller.RootController;
+import org.springframework.cloud.dataflow.server.controller.RuntimeModulesController;
+import org.springframework.cloud.dataflow.server.controller.RuntimeModulesController.AppInstanceController;
+import org.springframework.cloud.dataflow.server.controller.SecurityController;
+import org.springframework.cloud.dataflow.server.controller.StreamDefinitionController;
+import org.springframework.cloud.dataflow.server.controller.StreamDeploymentController;
+import org.springframework.cloud.dataflow.server.controller.TaskDefinitionController;
+import org.springframework.cloud.dataflow.server.controller.TaskDeploymentController;
+import org.springframework.cloud.dataflow.server.controller.TaskExecutionController;
+import org.springframework.cloud.dataflow.server.controller.UiController;
+import org.springframework.cloud.dataflow.server.repository.StreamDefinitionRepository;
+import org.springframework.cloud.dataflow.server.repository.TaskDefinitionRepository;
+import org.springframework.cloud.deployer.resource.maven.MavenProperties;
+import org.springframework.cloud.deployer.spi.app.AppDeployer;
+import org.springframework.cloud.deployer.spi.task.TaskLauncher;
+import org.springframework.cloud.stream.configuration.metadata.ModuleConfigurationMetadataResolver;
+import org.springframework.cloud.stream.module.metrics.FieldValueCounterRepository;
+import org.springframework.cloud.task.repository.TaskExplorer;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.hateoas.EntityLinks;
+
+/**
+ * Configuration for the Data Flow Server Controllers.
+ *
+ * @author Mark Fisher
+ */
+@Configuration
+@ConditionalOnBean({ AppDeployer.class, TaskLauncher.class })
+public class DataFlowControllerAutoConfiguration {
+
+	@Bean
+	public RootController rootController(EntityLinks entityLinks) {
```

Comment:
On yarn I'm building additional boot apps manually to get access to fresh beans with different configs. I got into trouble with `EntityLinks` because there we none to autowire. So it might be a good idea to add more conditions so that this autoconfig would not kick in if `EntityLinks` are not available. I got away from this problem by excluding this class. Trouble wasn't with a _main_ yarn dataflow server but with those extra `SpringApplication` instances I'm using which does same autoconfig magic behind a scenes.

It probably failed because I already excluded some other features like not enabling web meaning `hateoas` autoconfig probably didn't create anything.